### ISSUE TITLE
id_token in logout uri

### ISSUE THREAD

**Issue by ShyLionTjmn** [vouch/vouch-proxy#522]
My IdP requires id_token as one of parameters to be able to redirect to specified URL, like this:

`end_session_endpoint: https://idp.domain.com/oauth/logout?id_token_hint={ID_TOKEN}&post_logout_redirect_uri=https%3A%2F%2Fmyapp.domain.com%2F`

is there a way to include it in uri?

**Comment by bnfinet** [comment]
@ShyLionTjmn welcome back!

Which IdP is this?

VP does not support passing the token to the IdP in this fashion.

**Comment by aaronpk** [comment]
fwiw this is part of the OpenID Connect spec:

https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout

**Comment by bnfinet** [comment]
right now these URLs are listed at `vouch.post_logout_redirect_uris`
https://github.com/vouch/vouch-proxy/blob/master/config/config.yml_example#L194

If #41 OIDC Discovery from `.well-known` is implemented and VP is configured with the specified `end_session_endpoint` does the IdP include `id_token_hint={ID_TOKEN}`?

I don't see that mentioned here...
https://openid.net/specs/openid-connect-discovery-1_0.html

I'm thinking there may need to offer a new configuration parameter...
`vouch.post_logout_id_token_hint: true`

MicroSoft Azure chooses not to include `id_token_hint` when OIDC discovery is used.  
https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/active-directory-b2c/session-behavior.md

Seems like "Single Sign Out" is it's own rabbit hole. :)

**Comment by aaronpk** [comment]
> Seems like "Single Sign Out" is it's own rabbit hole. :)

It absolutely is :joy:

**Comment by ShyLionTjmn** [comment]
IdP is Blitz: 

https://blitz.mydomain.com/blitz/oauth/.well-known/openid-configuration has:
```"end_session_endpoint": "https://blitz.mydomain.com/blitz/oauth/logout",```

**Comment by ShyLionTjmn** [comment]
> I'm thinking there may need to be a new configuration parameter... `vouch.post_logout_id_token_hint: true` (default `false`)

that would be nice

**Comment by talasjanos** [comment]
I'd be really glad if this extra parameter could be implemented.

Keycloak also requires the `id_token_hint` parameter to skip the logout confirmation.

**Comment by talasjanos** [comment]
@ShyLionTjmn is this fixing your issue?

**Comment by ShyLionTjmn** [comment]
> 

didn't try it

**Comment by DenisNovac** [comment]
I don't think it works. Even when i have both tokens (i can see them in `vouch/validate`) it tries to do `oauth2/v1/logout` without `id_token_hint` which results in 400 for Okta because that parameter is required.

From the code in #258 it looks like it tries to pass `id_token_hint` from `id_token` claim which is `OAUTH_CLAIMS` env. But it didn't worked for me as well...

**Comment by DenisNovac** [comment]
Done more testing. It actually does work with this option:

```
VOUCH_HEADERS_IDTOKEN: X-Vouch-IdP-IdToken
```

However it only works like 50% of the time. Other times it gives an error:

```
{"level":"error","ts":1779434443.3162708,"msg":"token is malformed: token contains an invalid number of segments"}
```

I tried to check my it_token_hint through Developer Tools/JWT decode and it decodes fine. It has 3 segments as normal JWT.

With VOUCH_TESTING=true it gives one additional error on callback from Okta to Vouch:

```
x-vouch-error
/auth Invalid session state: stored %!s(<nil>), returned <REDACTED>
```
